src/vcpkg/archives.cpp

```diff
@@ -136,9 +136,23 @@ namespace
         {
             win32_extract_msi(archive, to_path);
         }
-        else if (Strings::case_insensitive_ascii_equals(ext, ".zip") ||
-                 Strings::case_insensitive_ascii_equals(ext, ".7z") ||
-                 Strings::case_insensitive_ascii_equals(ext, ".exe"))
+        else if (Strings::case_insensitive_ascii_equals(ext, ".zip"))
+        {
+            auto maybe_tar = paths.get_optional_tool_exe(Tools::TAR);
+            if (auto tar = maybe_tar.get())
+            {
+                extract_tar(*tar, archive, to_path);
+            }
+            else
+            {
+                win32_extract_with_seven_zip(paths, archive, to_path);
+            }
+        }
+        else if (Strings::case_insensitive_ascii_equals(ext, ".7z"))
+        {
+            extract_tar_cmake(paths.get_required_tool_exe(Tools::CMAKE), archive, to_path);
+        }
+        else if (Strings::case_insensitive_ascii_equals(ext, ".exe"))
```

Comment:
```suggestion
        else if (Strings::case_insensitive_ascii_equals(ext, ".7z.exe"))
```
7zip cannot extract random executables.

The reason cmake can probably not deal with it is that the installer part adds extra stuff which breaks the archive detection. So removing the extract stuff (somehow) could make normal extraction possible. Creating an installer is done by e.g. `copy /b 7zSD.sfx + config.txt + archive.7z archive.exe`. For git e.g. you can simply remove everything up until and including `;!@InstallEnd@!` from the downloaded blob and than simply run `cmake -E tar xf PortableGit-2.35.1.2-32-bit.7z` (Yeah that actually works, I tested it)

vcpkg could silent install if the sfx installer is `7zCon.sfx` (https://sevenzip.osdn.jp/chm/cmdline/switches/sfx.htm)
git for windows, however, uses a custom version of `7zSD.sfx` which will always pop/show the extraction progress (and close on finish).

Reply:
ext will never contain the two dotted thing (`.7z.exe`) you want to look for here since `extension()` of `"foo.7z.exe"` is `".exe"`.